lsl/common/mcs.py

```diff
@@ -955,17 +944,25 @@ def apply_pointing_correction(az, el, theta, phi, psi, lat=34.070, degrees=True)
     
     # Convert the az,alt coordinates to the unit vector
     if degrees:
-        xyz = aipy.coord.azalt2top((az*numpy.pi/180.0, el*numpy.pi/180.0))
+        az = az*astrounits.deg
+        el = el*astrounits.deg
     else:
-        xyz = aipy.coord.azalt2top((az, el))
-        
+        az = az*astrounits.rad
+        el = el*astrounits.rad
+    sph = SphericalRepresentation(az, el, 1)
+    xyz = sph.to_cartesian()
+    
     # Rotate
-    xyzP = numpy.dot(rot, xyz)
+    yxz = xyz.xyz[[1,0,2]]
+    xyzP = CartesianRepresentation((rot @ yxz)[[1,0,2]])
     
-    azP, elP = aipy.coord.top2azalt(xyzP)
+    sph = SphericalRepresentation.from_cartesian(xyzP)
     if degrees:
-        azP *= 180/numpy.pi
-        elP *= 180/numpy.pi
+        azP = sph.lon.deg
+        elP = sph.lat.deg
+    else:
+        azP = sph.lon.rad
+        elP = sph.lat.rad
```

Comment:
This is one area where we need to check to make sure that we are getting identical answers.

Reply:
Now tested against an old LSL version (2.1.7).